packages/backend/src/tokens/__tests__/request.test.ts

```diff
@@ -165,6 +165,181 @@ export default (QUnit: QUnit) => {
+      },
+      {
+        name: 'Can activate an org by ID with an unrelated path token in the prefix',
+        whenOrgSyncOptions: {
+          organizationPatterns: ['/unknown-thing/:any/orgs/:id'],
+        },
+        whenAppRequestPath: '/unknown-thing/thing/orgs/org_foo',
+        thenExpectActivationEntity: {
+          type: 'organization',
+          organizationId: 'org_foo',
+        },
+      },
+      {
+        name: 'Can activate an org by slug',
+        whenOrgSyncOptions: {
+          organizationPatterns: ['/orgs/:slug'],
+        },
+        whenAppRequestPath: '/orgs/my-org',
+        thenExpectActivationEntity: {
+          type: 'organization',
+          organizationSlug: 'my-org',
+        },
+      },
+      {
+        name: 'Can activate the personal workspace',
+        whenOrgSyncOptions: {
+          personalWorkspacePatterns: ['/personal-workspace'],
+        },
+        whenAppRequestPath: '/personal-workspace',
+        thenExpectActivationEntity: {
+          type: 'personalWorkspace',
+        },
+      },
+      {
+        name: 'ID match precedes slug match',
+        whenOrgSyncOptions: {
+          organizationPatterns: ['/orgs/:id', '/orgs/:slug'], // bad practice
+        },
+        whenAppRequestPath: '/orgs/my-org',
+        thenExpectActivationEntity: {
+          type: 'organization',
+          organizationId: 'my-org',
+        },
+      },
+      {
+        name: 'personal workspace match precedes org match',
+        whenOrgSyncOptions: {
+          organizationPatterns: ['/personal-workspace'], // bad practice
+          personalWorkspacePatterns: ['/personal-workspace'],
+        },
+        whenAppRequestPath: '/personal-workspace',
+        thenExpectActivationEntity: {
+          type: 'personalWorkspace',
+        },
+      },
+      {
+        name: 'personal workspace may contain path tokens',
+        whenOrgSyncOptions: {
+          personalWorkspacePatterns: ['/user/:any', '/user/:any/(.*)'],
+        },
+        whenAppRequestPath: '/user/123/home',
+        thenExpectActivationEntity: {
+          type: 'personalWorkspace',
+        },
+      },
+      {
+        name: 'All of the config at once',
+        whenOrgSyncOptions: {
+          organizationPatterns: [
+            '/orgs-by-id/:id',
+            '/orgs-by-id/:id/(.*)',
+            '/orgs-by-slug/:slug',
+            '/orgs-by-slug/:slug/(.*)',
+          ],
+          personalWorkspacePatterns: ['/personal-workspace', '/personal-workspace/(.*)'],
+        },
+        whenAppRequestPath: '/orgs-by-slug/org_bar/sub-resource',
+        thenExpectActivationEntity: {
+          type: 'organization',
+          organizationSlug: 'org_bar',
+        },
+      },
+    ];
+
+    testCases.forEach(testCase => {
+      test(testCase.name, assert => {
+        const path = new URL(`http://localhost:3000${testCase.whenAppRequestPath}`);
+        const toActivate = getOrganizationSyncTarget(path, testCase.whenOrgSyncOptions);
```

Comment:
To get this test to work, I had to export `getOrganizationSyncTarget` (see [here](https://github.com/clerk/javascript/pull/3977/files#diff-50844e9765ddaece7cd9885aa9a6b4556d93d6ccaf92ef2484399cec67a09405R642)), which I don't love. Any ideas on how to avoid that are welcome.